consumers/aws.go

```diff
@@ -54,17 +52,54 @@ func withClient(c AWSClient) *awsClient {
 }
 
 func (a *awsClient) Sync(endpoints []*pkg.Endpoint) error {
-	existingRecords, err := a.client.ListRecordSets()
+	newAliasRecords, err := a.client.EndpointsToAlias(endpoints)
 	if err != nil {
 		return err
 	}
 
-	recordInfoMap := a.client.RecordInfo(existingRecords)
-	newAliasRecords, err := a.client.EndpointsToAlias(endpoints)
+	hostedZonesMap, err := a.client.GetHostedZones()
+	if err != nil {
+		return err
+	}
+
+	inputByZoneID := map[string][]*route53.ResourceRecordSet{}
+	for _, record := range newAliasRecords {
+		zoneID := getZoneIDForEndpoint(hostedZonesMap, record) //this guarantees that the endpoint will not be created in multiple hosted zones
+		if zoneID == "" {
+			log.Infof("Hosted zone for endpoint: %s is not found. Skipping record...", record.Name)
```

Comment:
It doesn't sound like an Error if it's ok to skip like the message says.

I Think Warn would be better if it's considered a potential problem but you are able to continue.